mlir/lib/Dialect/DLTI/DLTI.cpp

```diff
@@ -375,32 +465,35 @@ class TargetDataLayoutInterface : public DataLayoutDialectInterface {
 };
 } // namespace
 
-void DLTIDialect::initialize() {
-  addAttributes<DataLayoutEntryAttr, DataLayoutSpecAttr>();
-  addInterfaces<TargetDataLayoutInterface>();
-}
-
-Attribute DLTIDialect::parseAttribute(DialectAsmParser &parser,
-                                      Type type) const {
-  StringRef attrKind;
-  if (parser.parseKeyword(&attrKind))
-    return {};
+namespace {
+class SystemDescSpecInterface : public DataLayoutDialectInterface {
```

Comment:
Document this please.

IIUC, this interface is only meant to run the verifier. This is rather unconventional as one typically wants to have access to the interface. Have you considered other approaches to trigger verification?